app/src/main/java/com/hover/stax/presentation/sim/SimScreen.kt

```diff
@@ -0,0 +1,477 @@
+                            item {
+                                LinkSimCard(id = R.string.link_sim_to_stax,
+                                    onClickedLinkSimCard = { })
+                            }
+                        }
+                    }
+
+                    itemsIndexed(accounts) { index, account ->
+                        if (account.id > 0) {
+                            SimItem(
+                                simIndex = 1,
+                                account = account,
+                                bonus = (0.05 * 100).toInt(),
+                                onClickedBuyAirtime = { },
+                                balanceTapListener = null
+                            )
+                        } else {
+                            UnSupportedSim(simInfo = null,
+                                slotId = 2,
+                                context = LocalContext.current)
+                        }
+                    }
+                }
+            })
+        }
+    }
+}
+
+@Composable
+private fun PageTitle() {
+    val size13 = dimensionResource(id = R.dimen.margin_13)
+    Text(
+        text = stringResource(id = R.string.your_linked_sim),
+        modifier = Modifier.padding(vertical = size13),
+        style = MaterialTheme.typography.button
+    )
+}
+
+@Composable
+private fun NoticeText(@StringRes stringRes: Int) {
+    val size13 = dimensionResource(id = R.dimen.margin_13)
+    val size34 = dimensionResource(id = R.dimen.margin_34)
+
+    Text(
+        text = stringResource(id = stringRes),
+        modifier = Modifier
+            .padding(vertical = size13, horizontal = size34)
+            .fillMaxWidth(),
+        textAlign = TextAlign.Center,
+        color = TextGrey,
+    )
+}
+
+@Composable
+private fun ShowGrantPermissionContent( onClickedAddNewAccount: () -> Unit ) {
+    Column {
+        NoticeText(stringRes = R.string.simpage_permission_alert)
+        LinkSimCard(
+            id = R.string.link_sim_to_stax,
+            onClickedLinkSimCard = onClickedAddNewAccount
+        )
+    }
+}
+
+@Composable
+private fun UnSupportedSim(simInfo: SimInfo?, slotId: Int, context: Context) {
+    val size13 = dimensionResource(id = R.dimen.margin_13)
+    var emailBody = ""
+    if(simInfo !=null) {
+        emailBody = stringResource(id = R.string.sim_card_support_request_emailBody,
+            simInfo.osReportedHni ?: "Null",
+            simInfo.operatorName ?: "Null",
+            simInfo.networkOperator ?: "Null",
+            simInfo.countryIso ?: "Null")
+    }
+
+    //Sometimes operator name comes in as : No service,
+    //We don't want user to see: "We detected a No service SIM in slot 1..."
+    //Instead it should be: "We detected a SIM in Slot 1", otherwise show "We detected a MTN SIM in Slot 1"
+    val actualOperatorName = simInfo?.operatorName ?: ""
+    val displayedNetworkName = if((actualOperatorName).contains("No service")) "" else actualOperatorName
```

Comment:
It is strange that operatorName is empty, that is the correct variable. I have reproduced the issue on my device/SIM will look into how to fix it. This is fine in the meantime, just want some minor UI changes which I will describe.